File: scripts/reset_skill.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-

"""
スキルファイルのリセットスクリプト

指定されたスキルファイル（Markdown形式）から、
「## ローカル・アダプテーション (Gemini固有)」というヘッダー以降の記述を削除し、
純粋な移植版の状態にリセットします。
"""

import sys
import os

# ローカル・アダプテーションセクションを識別するターゲットヘッダー
TARGET_HEADER = "## ローカル・アダプテーション (Gemini固有)"

def reset_skill_file(file_path: str) -> bool:
    """
    指定されたスキルファイルからローカル・アダプテーションセクションを削除します。

    Args:
        file_path (str): リセットするスキルファイルへのパス。

    Returns:
        bool: 正常にセクションが削除された場合はTrue、
              セクションが見つからないかファイルが存在しない場合はFalse。
    """
    # ファイルの存在確認
    if not os.path.exists(file_path):
        return False

    with open(file_path, 'r', encoding='utf-8') as f:
        lines = f.readlines()

    found_index = -1
    # ターゲットヘッダーの行番号を検索
    for i, line in enumerate(lines):
        if TARGET_HEADER in line:
            found_index = i
            break

    # ターゲットヘッダーが見つかった場合
    if found_index != -1:
        # ヘッダー以前の行を保持
        new_lines = lines[:found_index]
        
        # ヘッダー直前の不要な空行を削除し、クリーンな状態にする
        while new_lines and not new_lines[-1].strip():
            new_lines.pop()
        
        # ファイルの最後に改行があることを保証する（存在しない場合のみ追加）
        if new_lines and not new_lines[-1].endswith('\n'):
            new_lines[-1] += "\n"

        # 修正された内容でファイルを上書き保存
        with open(file_path, 'w', encoding='utf-8') as f:
            f.writelines(new_lines)
        return True
    else:
        # ターゲットヘッダーが見つからなかった場合
        return False

def main():
    """
    スクリプトのエントリーポイント。
    コマンドライン引数としてファイルパスを受け取り、reset_skill_fileを呼び出します。
    """
    if len(sys.argv) < 2:
        print("使用法: python reset_skill.py <スキルファイルのパス> ...")
        sys.exit(1)

    # 各ファイルパスに対してリセット処理を実行
    for path in sys.argv[1:]:
        reset_skill_file(path)

if __name__ == "__main__":
    main()

File: scripts/reset_skill.mjs
import fs from 'node:fs';
import path from 'node:path';
import { fileURLToPath } from 'node:url';

const TARGET_HEADER = "## ローカル・アダプテーション (Gemini固有)";

/**
 * 指定されたスキルファイルからローカル・アダプテーションセクションを削除します。
 *
 * @param {string} filePath - リセットするスキルファイルへのパス。
 * @returns {boolean} 正常にセクションが削除された場合は true、セクションが見つからないかファイルが存在しない場合は false。
 */
export function resetSkillFile(filePath) {
  if (!fs.existsSync(filePath)) return false;

  const content = fs.readFileSync(filePath, 'utf8');
  const index = content.indexOf(TARGET_HEADER);

  if (index === -1) return false;

  // セクションの直前までの内容を取得
  let newContent = content.slice(0, index);
  
  // 【知見】セクションを削除した後に不要な空行が残らないよう、trimEnd() で末尾の空白を削除します。
  // これにより、元のファイルでセクションの前に複数の空行が存在していた場合でも、
  // 後の改行付与ロジックと組み合わせて、適切に 1 つの改行に集約されます。
  newContent = newContent.trimEnd();
  
  // 【知見】POSIX 標準のテキストファイル形式（末尾改行）を維持するため、
  // コンテンツが存在する場合は末尾に必ず 1 つの改行を付与します。
  // newContent が空（ファイル全体が削除対象だった場合など）は、完全に空のファイルとします。
  if (newContent && !newContent.endsWith('\n')) {
    newContent += '\n';
  }

  fs.writeFileSync(filePath, newContent);
  return true;
}

/**
 * CLI のエントリポイント。
 */
function main() {
  const args = process.argv.slice(2);
  if (args.length === 0) {
    console.error("使用法: node scripts/reset_skill.mjs <SKILL_FILE_PATH> ...");
    process.exit(1);
  }

  let overallSuccess = true;

  for (const filePath of args) {
    try {
      const success = resetSkillFile(filePath);

      if (success) {
        console.log(`リセット成功: ${filePath}`);
      } else {
        console.error(`リセット失敗（セクション未検出またはファイル不在）: ${filePath}`);
        overallSuccess = false;
      }
    } catch (error) {
      console.error(`予期せぬエラーが発生しました (${filePath}): ${error.message}`);
      overallSuccess = false;
    }
  }

  if (!overallSuccess) {
    process.exit(1);
  }
}

// エントリポイントのガード
const scriptPath = fileURLToPath(import.meta.url);
const isMain = process.argv[1] && (
  process.argv[1] === scriptPath || 
  path.resolve(process.argv[1]) === scriptPath
);

if (isMain) {
  main();
}
